### PR TITLE
Ui fixes 2

### PR DESCRIPTION
**Pull request by soofstad**
- More input validation, user feedback, and error handling
- Don't display empty combinations

### CODE REVIEW

File: web/src/Components/Optimization/PillInput.tsx [hunk 1]
```diff
@@ -1,79 +1,104 @@
...
+    let newValue: number = 0
+    if (value !== '') newValue = parseInt(value)
 
-        setPill({
-            volume: volume,
-            mass: mass,
-            density: pill.density
-        })
+    if (Math.sign(newValue) <= 0) {
+      type === 'volume' && setInvalidVolume(true)
```
Comment (eoaksnes) [inline web/src/Components/Optimization/PillInput.tsx:38]: If you set wrong (invalid) volume, then density is also shown wrong? However, density could be an ok value?

File: web/src/Components/Optimization/PillInput.tsx [hunk 2]
```diff
@@ -1,79 +1,104 @@
...
-            density: density
-        })
+    if (type === 'volume') {
+      setPill({
+        volume: newValue,
+        mass: newValue * pill.density,
+        density: pill.density,
+      })
+    }
+    if (type === 'density') {
```
Comment (eoaksnes) [inline web/src/Components/Optimization/PillInput.tsx:52]: If we want to use same function for updating both volume and density, please make enum for checking if it is volume or density.

File: web/src/Components/Optimization/PillInput.tsx (hunk 1 above)
Comment (eoaksnes) [reply web/src/Components/Optimization/PillInput.tsx:38]: Mass should be shown as invalid, but not sure that density should.

Comment (soofstad) [reply web/src/Components/Optimization/PillInput.tsx:38]: Only the invalid property is shown as invalid.
Mass is not displayed.
User cant run optimization with invalid values.

File: web/src/Components/Optimization/PillInput.tsx (hunk 2 above)
Comment (soofstad) [reply web/src/Components/Optimization/PillInput.tsx]: Fixed

**Pull request merged**